### PR TITLE
train_bc script and associated changes to BC

### PR DESCRIPTION
**Pull request by shwang**
Adds a Sacred script for BC.

Other changes:
* Adds to BC.train the `log_rollouts_venv` and `log_rollouts_n_episodes` parameters, which can be used to log rollout statistics every `log_interval` updates.
  * I used these parameters for experiments earlier, but perhaps a callback would be the better choice here
* BC.save_policy: use `AnyPath` type instead of `str`

### CODE REVIEW

File: src/imitation/algorithms/bc.py
Comment (AdamGleave) [inline src/imitation/algorithms/bc.py:380]: This does seem like it might be a better solution. Does [EvalCallback](https://stable-baselines3.readthedocs.io/en/master/guide/callbacks.html#evalcallback) do what you need? It does log [episode reward and length](https://stable-baselines3.readthedocs.io/en/master/_modules/stable_baselines3/common/callbacks.html).

The flip-side is I think every other algorithm we have logs stuff like this without a callback? So perhaps it does make sense this is built into BC.

File: src/imitation/scripts/train_bc.py
Comment (AdamGleave) [inline src/imitation/scripts/train_bc.py:67]: `assert` is meant for internal consistency checks, not input validation; note they will be disabled when running with `-O`. raise ValueError instead if missing.

File: src/imitation/scripts/train_bc.py
Comment (AdamGleave) [inline src/imitation/scripts/train_bc.py:77]: Redundant -- you check and raise ValueError later (good!)

File: src/imitation/scripts/train_bc.py
Comment (AdamGleave) [inline src/imitation/scripts/train_bc.py:87]: Should be a TODO comment?

File: src/imitation/scripts/train_bc.py
Comment (AdamGleave) [inline src/imitation/scripts/train_bc.py:122]: Also should be a TODO comment.

File: tests/test_scripts.py
Comment (AdamGleave) [inline tests/test_scripts.py:64]: This fails because the data file is missing -- you need to set the path to `CARTPOLE_TEST_POLICY_PATH`.

**Review by AdamGleave — APPROVED** [review]
Some minor changes needed but otherwise LGTM. Tests failing due to path error that should be fixed before merge.

File: src/imitation/scripts/train_bc.py
Comment (shwang) [reply src/imitation/scripts/train_bc.py:67]: Makes sense, thanks

File: tests/test_scripts.py
Comment (shwang) [reply tests/test_scripts.py:64]: ah that's right. Fixed!

File: src/imitation/algorithms/bc.py
Comment (shwang) [reply src/imitation/algorithms/bc.py:382]: EvalCallback does seem like a good fit -- adding it to the comment here. For better SB3 integration it could be nice to add Callbacks to every imitation alg's `train` someday.

**Pull request merged**
